src/main/java/org/folio/innreach/domain/service/impl/AgencyServiceImpl.java

```
@@ -0,0 +1,95 @@
+import org.folio.innreach.dto.AgenciesPerCentralServerDTO;
+import org.folio.innreach.dto.Agency;
+import org.folio.innreach.dto.CentralServerAgenciesDTO;
+import org.folio.innreach.dto.CentralServerDTO;
+import org.folio.innreach.dto.InnReachResponseDTO;
+import org.folio.innreach.dto.LocalServerAgenciesDTO;
+import org.folio.innreach.external.exception.InnReachException;
+import org.folio.innreach.external.service.InnReachExternalService;
+
+@Log4j2
+@RequiredArgsConstructor
+@Service
+public class AgencyServiceImpl implements AgencyService {
+
+  private static final String INN_REACH_LOCAL_SERVERS_URI = "/contribution/localservers";
+
+  private final CentralServerService centralServerService;
+  private final InnReachExternalService innReachService;
+
+
+  @Override
+  @Transactional(readOnly = true)
+  public CentralServerAgenciesDTO getAllAgencies() {
+    var servers = centralServerService.getAllCentralServers(0, Integer.MAX_VALUE).getCentralServers();
+
+    var agencies = mapItemsWithFilter(servers, this::retrieveAllAgenciesFromCentralServer, Objects::nonNull);
+
+    return new CentralServerAgenciesDTO()
+        .centralServerAgencies(agencies)
+        .totalRecords(agencies.size());
+  }
+
+  private AgenciesPerCentralServerDTO retrieveAllAgenciesFromCentralServer(CentralServerDTO centralServer) {
+    log.info("Retrieving local server agencies from central server: code = {}", centralServer.getCentralServerCode());
+
+    LocalServerAgenciesDTO lsaResponse;
+    try {
+      String response = innReachService.callInnReachApi(centralServer.getId(), INN_REACH_LOCAL_SERVERS_URI);
+
+      lsaResponse = fromJson(response, LocalServerAgenciesDTO.class);
+    } catch (InnReachException | BadCredentialsException | IllegalStateException e) {
+      log.warn("Failed to get local server agencies from central server: code = {}",
+          centralServer.getCentralServerCode(), e);
+
+      return null;
```

if we want to ignore the error completely then AgenciesPerCentralServerDTO with empty list will suit better

---

it won't. if you take a look at the method above, there is a standard filter to remove `null` values from the result. With empty `AgenciesPerCentralServerDTO` it'd be needed to provide custom filter.
`null`s are perfectly fine here